adapters/impactify/impactify.go

```diff
@@ -0,0 +1,186 @@
+package impactify
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"strings"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type ImpactifyExtBidder struct {
+	Impactify openrtb_ext.ExtImpImpactify `json:"impactify"`
+}
+
+type DefaultExtBidder struct {
+	Bidder openrtb_ext.ExtImpImpactify `json:"bidder"`
+}
+
+func (a *adapter) MakeRequests(bidRequest *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb2.Imp
+
+	var adapterRequests []*adapters.RequestData
+
+	for i := 0; i < len(bidRequest.Imp); i++ {
+		// Check if imp comes with bid floor amount defined in a foreign currency
+		if bidRequest.Imp[i].BidFloor > 0 && bidRequest.Imp[i].BidFloorCur != "" && strings.ToUpper(bidRequest.Imp[i].BidFloorCur) != "USD" {
+			// Convert to US dollars
+			convertedValue, err := reqInfo.ConvertCurrency(bidRequest.Imp[i].BidFloor, bidRequest.Imp[i].BidFloorCur, "USD")
+			if err != nil {
+				return nil, []error{err}
+			}
+			bidRequest.Imp[i].BidFloorCur = "USD"
+			bidRequest.Imp[i].BidFloor = convertedValue
+		}
+
+		var impactifyExt ImpactifyExtBidder
+
+		var defaultExt DefaultExtBidder
+		json.Unmarshal(bidRequest.Imp[i].Ext, &defaultExt)
+		impactifyExt.Impactify = defaultExt.Bidder
+		bidRequest.Imp[i].Ext, _ = json.Marshal(impactifyExt)
```

Comment:
Please do not ignore the error from `json.Marshal`.

Reply:
Yes. The conditions has been added.